### DISCUSSION TITLE
Trouble with PauliSigma

### DISCUSSION THREAD

**Discussion by ammohseni** [FeynCalc/feyncalc#75] [Q&A] [answered]
Hello, 

I have a model which I implemented using FeynRules and works very well with FeynArts but once FeynArts is patched with FeynCalc, upon running "InsertFields" I get

PauliSigma[{SU2W$1, 1, 1}] called with 3 arguments; 1 or 2 arguments are expected

I will be thankful if you give me an idea of what the problem may be.Thanks in advance.

Regards.

**Comment by vsht** [comment, marked as answer]
Hi,

PauliSigma is also a FeynCalc symbol
https://feyncalc.github.io/FeynCalcBook/ref/PauliSigma.html
with a specific definition that is not compatible with your model.

However, I'm rather wondering why does your FeynArts model contain PauliSigma at all?
If you look at FeynRules/Models/SM/SM.fr, you see that the sigmas appear only in the FeynRules model
but the final generated FeynArts model contains no such objects, since all the corresponding expressions
are rewritten according to the definitions

```
PauliSigma[1] = {{0,1},{1,0}};
PauliSigma[2] = {{0,-I},{I,0}};
PauliSigma[3] = {{1,0},{0,-1}};
```

Cheers,
Vladyslav

**Comment by ammohseni** [comment]
Thank you for your answer, I have defined Pauli matrices as SU2 generators in FeynRules model file (although that is not necessary). So based on what you said it is probably interfering with FeynCalc's PauliSigmas. I'll remove that. 

Best regards.

**Comment by vsht** [comment]
Sure, Pauli matrices appear in the Lagrangian acting on the 2-vectors of fields. 
But for explicit calculations you usually write them out explicitly, so that the Feynman rules contain
no Pauli matrices. FeynRules allows you to specify components of the generator matrices and this
way you can create a proper FeynArts model that will involve no PauliSigma objects.
